src/Appwrite/Auth/OAuth2/Notion.php

```diff
@@ -51,32 +56,25 @@ public function getLoginURL():string
     /**
      * @param string $code
      *
-     * @return string
+     * @return array
      */
-    public function getAccessToken(string $code):string
+    public function getTokens(string $code): array
     {
-        $headers = [
-            "Authorization: Basic " . \base64_encode($this->appID . ":" . $this->appSecret),
-        ];
-
-        $response = $this->request(
-            'POST',
-            $this->endpoint . '/oauth/token',
-            $headers,
-            \http_build_query([
-                'grant_type' => 'authorization_code',
-                'redirect_uri' => $this->callback,
-                'code' => $code
-            ])
-        );
-
-        $response = \json_decode($response, true);
-
-        if (isset($response['access_token'])) {
-            return $response['access_token'];
+        if(empty($this->tokens)) {
+            $headers = ['Authorization: Basic ' . \base64_encode($this->appID . ':' . $this->appSecret)];
```

Comment:
Should building the `Authorization Basic` header become a function for reuse?

Reply:
It could, for this PR I re-used what was already there. We could make new PR with these improvements, I also saw a lo of potential for re-use. 90% of providers follow protocol strictly and only differs in path.